frontend/src/components/common/StatusFilter.tsx

```diff
@@ -0,0 +1,141 @@
+import { Icon } from '@iconify/react';
+import Autocomplete from '@mui/material/Autocomplete';
+import Box from '@mui/material/Box';
+import Checkbox from '@mui/material/Checkbox';
+import ListItem from '@mui/material/ListItem';
+import { useTheme } from '@mui/material/styles';
+import TextField from '@mui/material/TextField';
+import Typography from '@mui/material/Typography';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { useHistory, useLocation } from 'react-router-dom';
+import { addQuery } from '../../helpers';
+import { setStatusFilter } from '../../redux/filterSlice';
+import { useTypedSelector } from '../../redux/reducers/reducers';
+
+export interface PureStatusesAutocompleteProps {
+  statusNames: string[];
+  onChange: (event: React.ChangeEvent<{}>, newValue: string[]) => void;
+  filter: { statuses: Set<string>; search: string };
```

Comment:
I think if you made this `filter?:` and change the rest of the code to account for that some of the tests would pass. Because currently they are sending filter as undefined.